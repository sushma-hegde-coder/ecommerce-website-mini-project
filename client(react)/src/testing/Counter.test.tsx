import { render, screen } from "@testing-library/react";
import Counter from "./Counter";
import UserEvent from "@testing-library/user-event";

describe("Counter Component", () => {
  test("on load it should display 0", () => {
    render(<Counter />);
    // assert functions
    expect(screen.getByTestId("countervalue")).toHaveTextContent("0");
  });
  test("on increment click it should display 1", () => {
    render(<Counter />);
    // simulate event
    UserEvent.click(screen.getByTestId("increment-btn"));
    // assert functions
    expect(screen.getByTestId("countervalue")).toHaveTextContent("1");
  });
  test("on decrement click it should be -1", () => {
    render(<Counter />);
    // simulate event
    UserEvent.click(screen.getByTestId("decrement-btn"));
    // assert functions
    expect(screen.getByTestId("countervalue")).toHaveTextContent("-1");
  });
  test("match snapshot", () => {
    const { asFragment } = render(<Counter />);
    expect(asFragment).toMatchSnapshot();
  });
});
