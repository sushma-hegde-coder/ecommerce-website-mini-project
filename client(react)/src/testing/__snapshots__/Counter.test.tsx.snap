// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component match snapshot 1`] = `[Function]`;
